src/ExportMap.js

```diff
@@ -43,6 +43,13 @@ export default class ExportMap {
      */
     this.imports = new Map();
     this.errors = [];
+    /**
+     * true when we are still unsure if
+     * the module is ESM, happens when the module
+     * contains dynamic `import()` but no other
+     * ESM import/export
+     */
+    this.maybeNotEsm = false;
```

Comment:
"containing dynamic import" tells us nothing, so it'd be find to maybe name this `this.parseGoal`?

that way it could be set to `'ambiguous'` by default, `'Module'` when it finds an export or an import, `Script` when it finds a `with` statement, etc.

Reply:
Sounds reasonable. Basically make the three types of results explicit.

Reply:
But what happens in the cases where the code returns a `null` `ExportMap` currently? Should  it return an `ExportMap` with `parseGoal` set to `Script` as a sort of null object?

Reply:
That seems reasonable - to always have a consistent type returned.